src/Routes/Devices/DeviceTable.js

```diff
@@ -62,10 +61,10 @@ const columnNames = [
   },
 ];
 
-const createRows = (devices, hasLinks, fetchDevices) => {
+const createRows = (devices, hasLinks, fetchDevices, groupName) => {
+  const location = useLocation();
```

Comment:
```suggestion
const createRows = (devices, hasLinks, fetchDevices, groupName, history) => {
```

Reply:
In general, hooks should be called only at the beginning of component functions, not in other functions.